crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -301,7 +301,11 @@ impl Provider for KvRedisProvider {
             source_id, config, ..
         }: LinkConfig<'_>,
     ) -> anyhow::Result<()> {
-        let conn = if let Some(url) = config.get(CONFIG_REDIS_URL_KEY) {
+        let conn = if let Some(url) = config
+            .keys()
+            .find(|k| k.eq_ignore_ascii_case(CONFIG_REDIS_URL_KEY))
```

Comment:
but but this is O(n) not O(1)! What if I have millions of keys!?!?!?

Reply:
I honestly did/am considering doing a check for both the original key and it lowercased... O(2) so much better than O(n), but we had this code "laying around" so to speak...